detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/process/GoReplacementManager.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detectable.detectables.go.gomod.process;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.bdio.model.dependency.Dependency;
+import com.synopsys.integration.bdio.model.externalid.ExternalIdFactory;
+import com.synopsys.integration.detectable.detectables.go.gomod.model.GoListAllData;
+import com.synopsys.integration.detectable.detectables.go.gomod.model.ReplaceData;
+
+public class GoReplacementManager {
+    private static final String INCOMPATIBLE_SUFFIX = "+incompatible";
+
+    private final ExternalIdFactory externalIdFactory;
+
+    private final List<GoListAllData> allModules;
+
+    public GoReplacementManager(List<GoListAllData> allModules, ExternalIdFactory externalIdFactory) {
+        this.allModules = allModules;
+        this.externalIdFactory = externalIdFactory;
+    }
+
+    public Dependency getReplacementForModule(String moduleName) {
+        String replacementName = getPathForModule(moduleName).orElse(moduleName);
+        String replacementVersion = getVersionForModule(moduleName).orElse(null);
+        return new Dependency(replacementName, replacementVersion, externalIdFactory.createNameVersionExternalId(Forge.GOLANG, moduleName, replacementVersion));
+    }
+
+    public Optional<String> getPathForModule(String moduleName) {
+        return allModules.stream()
```

Comment:
My implementation was already iffy with streaming over all modules for every dependency, but now we do it twice 😢 
Perhaps we should construct a mapping of `String` module name to replaced `Dependency` in the constructor.